python/mspasspy/db/database.py

```diff
@@ -269,6 +282,16 @@ def read_data(self, object_id, mode='promiscuous', normalize=None, load_history=
           for save and read operations to improve the efficiency and simplify
           read operations.   Default turns this off by setting the tag null (None).
         :type data_tag: :class:`str`
+        :param alg_name: alg_name is the name the func we are gonna save while preserving the history.
+        :type alg_name: :class:`str`
+        :param alg_id: alg_id is a unique id to record the usage of func while preserving the history.
+        :type alg_id: :class:`bson.objectid.ObjectId`
+        :param define_as_raw: a boolean control whether we would like to set_as_origin when loading processing history
+        :type define_as_raw: :class:`bool`
+        :param retrieve_history_record: a boolean control whether we would like to load processing history
+        :type retrieve_history_record: :class:`bool`
+        :param fill_value: Fill value for gaps. Defaults to None. Traces will be converted to NumPy masked arrays if no value is given and gaps are present.
```

Comment:
"Traces will be converted to NumPy masked arrays if no value is given and gaps are present" I don't think this is true. This is what obspy does, but in our data object we don't have such feature.